contracts/src/TokenLib.sol

```diff
@@ -6,178 +6,48 @@ pragma solidity 0.8.28;
 
 import {IERC20} from "./interfaces/IERC20.sol";
 import {IERC20Permit} from "./interfaces/IERC20Permit.sol";
+import {ECDSA} from "openzeppelin/utils/cryptography/ECDSA.sol";
 
 library TokenLib {
```

Comment:
Do we still gain anything from having a separate token library? This will get inlined into the Token.sol? So is it worth the extra indirection and mental overhead?

Reply:
It gets inlined into `TokenLib.sol`, which is shared by instances of `Token.sol`.